app/javascript/utilities/http/request.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Generic request with all the default headers required by the application.
+ *
+ * @example
+ * import { request } from '@utilities/http';
+ *
+ * const response = await request('/notification_subscriptions/Article/26')
+ *
+ * Note:
+ * The body option will typically be passed in as a JavaScript object.
+ * A check is performed for this and automatically convert it to JSON if necessary.
+ *
+ * Requests send JSON by default but this can be easily overridden by adding
+ * the Accept and Content-Type headers to the request options.
+ *
+ * The default method is GET.
+ *
+ * @param {string} url The URL to make the request to.
+ * @param {RequestInit} [options={}] The request options.
+ *
+ * @return {Promise<Response>} the response
+ */
+export async function request(url, options = {}) {
+  const {
+    headers,
+    body,
+    method = 'GET',
+    csrfToken = window.csrfToken,
```

Comment:
Since on cached pages we load `csrfTokens` asynchronously, the `window.csrfToken` may not be available _immediately_. It almost always is available by the time we make a request but it's not guaranteed.

We deal with this in a few kind of hacky ways. If we're committing to a utility class we might want to try and come up with a way to somewhat ensure we wait for this availability without otherwise complicating things.

Reply:
I found the utility function to call `getCsrfToken()`. This should be good to go now.